amundsen_application/static/js/components/SearchPage/index.tsx

```diff
@@ -42,57 +42,59 @@ import {
 
 export interface DispatchFromProps {
-  searchAll: (term: string, options?: SearchAllOptions) => SearchAllRequest;
+  searchAll: (term: string, selectedTab: ResourceType, pageIndex: number) => SearchAllRequest;
   searchResource: (resource: ResourceType, term: string, pageIndex: number) => SearchResourceRequest;
+  updateSearchTab: (selectedTab: ResourceType) => UpdateSearchTabRequest;
 }
 
 export type SearchPageProps = StateFromProps & DispatchFromProps & RouteComponentProps<any>;
 
-interface SearchPageState {
-  selectedTab: ResourceType;
-}
-
-export class SearchPage extends React.Component<SearchPageProps, SearchPageState> {
+export class SearchPage extends React.Component<SearchPageProps> {
   public static defaultProps: Partial<SearchPageProps> = {};
 
   constructor(props) {
     super(props);
-
-    this.state = {
-      selectedTab: ResourceType.table,
-    };
   }
 
   componentDidMount() {
     const params = qs.parse(this.props.location.search);
     const { searchTerm, pageIndex, selectedTab } = params;
-    const { term, index, currentTab } = this.getSanitizedUrlParams(searchTerm, pageIndex, selectedTab);
-    this.setState({ selectedTab: currentTab });
+    const { term, index, tab } = this.getSanitizedUrlParams(searchTerm, pageIndex, selectedTab);
+
+    this.props.updateSearchTab(tab);
+
     if (term !== '') {
-      this.props.searchAll(term, this.createSearchOptions(index, currentTab));
-      if (currentTab !== selectedTab || pageIndex !== index) {
-        this.updatePageUrl(term, currentTab, index);
+      this.props.searchAll(term, tab, index);
+      if (tab !== selectedTab || pageIndex !== index) {
+        this.updatePageUrl(term, tab, index);
       }
     }
   }
 
   componentDidUpdate(prevProps) {
-    if (this.props.location.search !== prevProps.location.search) {
-      const params = qs.parse(this.props.location.search);
-      const { searchTerm, pageIndex, selectedTab } = params;
-      const { term, index, currentTab } = this.getSanitizedUrlParams(searchTerm, pageIndex, selectedTab);
-      this.setState({ selectedTab: currentTab });
-      const prevTerm = prevProps.searchTerm;
-      if (term !== prevTerm) {
-        this.props.searchAll(term, this.createSearchOptions(index, currentTab));
-      }
+    const searchParams = qs.parse(this.props.location.search);
+    const { searchTerm, pageIndex, selectedTab } = searchParams;
+    const { term, index, tab } = this.getSanitizedUrlParams(searchTerm, pageIndex, selectedTab);
+
+    const prevParams = qs.parse(prevProps.location.search);
+
+    // New search entered
+    if (term !== prevParams.searchTerm) {
+      this.props.searchAll(term, tab, index);
+    // Tab change
+    } else if (tab !== this.props.selectedTab) {
+      this.props.updateSearchTab(tab);
+    // Pagination change
+    } else if (prevParams.pageIndex !== undefined && index !== prevParams.pageIndex) {
+      this.props.searchResource(this.props.selectedTab, this.props.searchTerm, index);
```

Comment:
Simplified the update pattern for this component: user actions first update the page URL, which is then parsed and updates the internal state. This also fixes an issue where the URL and global state were getting out of sync (browser back to change pagination).